cloud/scope/powervs_cluster_test.go

```diff
@@ -1704,3 +1706,328 @@ func TestCreateServiceInstance(t *testing.T) {
+		}
+
+		dhcpServer := &models.DHCPServerDetail{ID: ptr.To("dhcpID"), Status: ptr.To(string(infrav1beta2.DHCPServerStateActive))}
+		mockPowerVS.EXPECT().GetDHCPServer(gomock.Any()).Return(dhcpServer, nil)
+
+		requeue, err := clusterScope.ReconcileNetwork()
+		g.Expect(err).To(BeNil())
+		g.Expect(requeue).To(BeTrue())
+	})
+	t.Run("When DHCPID is empty and GetNetworkByID returns error ", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMPowerVSClient: mockPowerVS,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{Spec: infrav1beta2.IBMPowerVSClusterSpec{
+				Network: infrav1beta2.IBMPowerVSResourceReference{ID: ptr.To("networkID")}}},
+		}
+		network := &models.Network{}
+		mockPowerVS.EXPECT().GetNetworkByID(gomock.Any()).Return(network, fmt.Errorf("GetNetworkByID error"))
+
+		requeue, err := clusterScope.ReconcileNetwork()
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(requeue).To(BeFalse())
+	})
+	t.Run("When DHCPID is empty and networkID is not empty", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+		network := &models.Network{NetworkID: ptr.To("networkID")}
+		clusterScope := PowerVSClusterScope{
+			IBMPowerVSClient: mockPowerVS,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{Spec: infrav1beta2.IBMPowerVSClusterSpec{
+				Network: infrav1beta2.IBMPowerVSResourceReference{ID: ptr.To("networkID")}}},
+		}
+		mockPowerVS.EXPECT().GetNetworkByID(gomock.Any()).Return(network, nil)
+		requeue, err := clusterScope.ReconcileNetwork()
+		g.Expect(clusterScope.IBMPowerVSCluster.Status.Network.ID).To(Equal(network.NetworkID))
+		g.Expect(err).To(BeNil())
+		g.Expect(requeue).To(BeTrue())
+	})
+	t.Run("When DHCPID is empty and checkNetwork returns empty network ID", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		network := &models.Network{}
+		dhcpNetwork := &models.DHCPServerNetwork{ID: ptr.To("dhcpNetworkID")}
+		dhcpServer := &models.DHCPServer{ID: ptr.To("dhcpID"), Network: dhcpNetwork}
+		clusterScope := PowerVSClusterScope{
+			IBMPowerVSClient: mockPowerVS,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{Spec: infrav1beta2.IBMPowerVSClusterSpec{
+				Network: infrav1beta2.IBMPowerVSResourceReference{ID: ptr.To("networkID")}}},
+		}
+		mockPowerVS.EXPECT().GetNetworkByID(gomock.Any()).Return(network, nil)
+		mockPowerVS.EXPECT().CreateDHCPServer(gomock.Any()).Return(dhcpServer, nil)
+		requeue, err := clusterScope.ReconcileNetwork()
+		g.Expect(clusterScope.IBMPowerVSCluster.Status.DHCPServer.ID).To(Equal(dhcpServer.ID))
+		g.Expect(clusterScope.IBMPowerVSCluster.Status.Network.ID).To(Equal(dhcpNetwork.ID))
+		g.Expect(err).To(BeNil())
+		g.Expect(requeue).To(BeFalse())
+	})
+	t.Run("When DHCPID and networkID is not empty and createDHCPServer returns error", func(t *testing.T) {
```

Comment:
similar to above this case will need modification.

Reply:
Addressed